rasa/core/events/__init__.py

```diff
@@ -347,13 +348,8 @@ def message(self) -> Dict[Text, Any]:
 
         m = self.data.copy()
         m["text"] = self.text
+        m.update(self.metadata.copy())
```

Comment:
copy isn't necessary, it will create a shallow copy anyways which we don't need

```suggestion
        m.update(self.metadata)
```

Reply:
Got it -- realized we only do it the first time because we're creating m 👍